File: u03/src/Binaer.java
import java.util.Scanner;

/*
 * Dieses Programm gibt die Binaerdarstellung einer positiven Zahl aus, ohne Arrays oder String-
 * Operationen.
 */
public class Binaer {

	public static void main(String[] args) {
		Scanner myConsole = new Scanner(System.in);
		int num = retrieveInt("Enter a number:", myConsole);

		printBinary(num);
	}

	public static int retrieveInt(String msg, Scanner input) {
		System.out.println(msg);
		try {
			return Integer.parseInt(input.nextLine());
		} catch (NumberFormatException e) {
			return retrieveInt("Please enter a valid number.", input);
		}
	}

	public static void printBinary(int num) {
		if (num <= 0) return;
		printBinary(num/2);
		System.out.print(num%2);
	}
}

File: u03/src/Binaer.cpp
#include <iostream>
using namespace std;

int retrieveInt(string prompt)
{
    cout << prompt << endl;
    string ans;
    cin >> ans;
    try
    {
        return stoi(ans);
    }
    catch (exception e)
    {
        return retrieveInt("Please enter a valid number:");
    }
}

void printBinary(int num)
{
    int digit = num % 2;
    if (num > 1)
    {
        printBinary(num / 2);
    }
    cout << digit;
}

int main()
{
    int num = retrieveInt("Enter a number:");

    printBinary(num);
    cout << endl;

    return 0;
}